route/src/route_generator_worker.cpp

```diff
@@ -377,14 +377,19 @@ namespace route {
                 publish_route_event(cav_msgs::RouteEvent::ROUTE_DEPARTED);
             }
             // check if we reached our destination be remaining down track distance
-            if(current_downtrack_distance_ > world_model_->getRoute()->length2d() - down_track_target_range_)
+            if(current_downtrack_distance_ > world_model_->getRoute()->length2d() - down_track_target_range_ && current_speed_ == 0 || current_downtrack_distance_ > world_model_->getRoute()->length2d())
```

Comment:
Comparing current_speed as a double to 0 is potentially complicated, floating point precision is a tricky thing. I'd recommend the more robust `current_speed < epsilon` where epsilon is a small (0.001) constant.

Reply:
I'm also not 100% sure this works as intended without additional parenthesis around the first "... & ..." clause. It short circuit the evaluation of the whole expression of either of those terms is not-true, and then it wouldn't even check the or-clause. I'm not 100% sure, but parens would make it explicit.